app/controllers/api/projects.php

```diff
@@ -1016,6 +1016,14 @@
         /** @var Appwrite\Utopia\Response $response */
         /** @var Utopia\Database\Database $dbForConsole */
 
+        // Ensure hostname has proper structure (no port, protocol..)
```

Comment:
Why don't we initialize this param in the route signature instead of the Text validator?

Reply:
We nest it to get both checks. FIrst, check for size (max length 256), then format (port, protocol,...)

Reply:
We used to have this validator just for cases like this: https://github.com/utopia-php/framework/blob/fix-for-0.4/src/Validator/Multiple.php - not sure why we deprecate it. It could be very useful and help us stay much more consistent in this case and take advantage of the framework proper lifecycle: `input -> validation -> execution`

Reply:
Thinking about it, we might have deprecated it because it was hard to generate a proper type for it in the APIs specs. Another option could be to add a hostname default length.

Some resources:
* https://en.wikipedia.org/wiki/Hostname#:~:text=The%20entire%20hostname%2C%20including%20the,maximum%20of%20253%20ASCII%20characters.
* https://stackoverflow.com/questions/8724954/what-is-the-maximum-number-of-characters-for-a-host-name-in-unix
<img width="731" alt="image" src="https://user-images.githubusercontent.com/1297371/167236952-294c9dcf-6f65-4f6e-944a-14773a24731b.png">

Reply:
Addressed in a PR: https://github.com/utopia-php/framework/pull/59/files